### ISSUE TITLE
Vertx Context Missing During Startup for Tracing

### ISSUE THREAD

**Issue by hjed** [eclipse-vertx/vertx-tracing#62]
When starting the application with tracing and micrometer metrics (+ opencensus bridge) it fails because the vert.x implementation of open telemetry require a current context and there isn't one during startup.

### Version
Vertx 4.4.4 (tried with OpenTelemtry versions 1.24 and 1.29) 

### Context
This is the exception:

```
unable to create stackdriver client Caused by: NullPointerException: Cannot invoke "io.vertx.core.Context.getLocal(Object)" because "vertxCtx" is null
	at io.opentelemetry.context.Context.makeCurrent(Context.java:202)
	at io.opentelemetry.opencensusshim.OpenTelemetryCtx.attach(OpenTelemetryCtx.java:30)
	at io.opencensus.trace.CurrentSpanUtils$ScopeInSpan.<init>(CurrentSpanUtils.java:95)
	at io.opencensus.trace.CurrentSpanUtils$ScopeInSpan.<init>(CurrentSpanUtils.java:80)
	at io.opencensus.trace.CurrentSpanUtils.withSpan(CurrentSpanUtils.java:52)
	at io.opencensus.trace.Tracer.withSpan(Tracer.java:154)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1009)
	at com.google.auth.oauth2.ComputeEngineCredentials.pingComputeEngineMetadata(ComputeEngineCredentials.java:371)
	at com.google.auth.oauth2.ComputeEngineCredentials.isOnGce(ComputeEngineCredentials.java:320)
	at com.google.auth.oauth2.DefaultCredentialsProvider.tryGetComputeCredentials(DefaultCredentialsProvider.java:313)
	at com.google.auth.oauth2.DefaultCredentialsProvider.getDefaultCredentialsUnsynchronized(DefaultCredentialsProvider.java:216)
	at com.google.auth.oauth2.DefaultCredentialsProvider.getDefaultCredentials(DefaultCredentialsProvider.java:119)
	at com.google.auth.oauth2.GoogleCredentials.getApplicationDefault(GoogleCredentials.java:129)
	at com.google.auth.oauth2.GoogleCredentials.getApplicationDefault(GoogleCredentials.java:101)
	at com.google.api.gax.core.GoogleCredentialsProvider.getCredentials(GoogleCredentialsProvider.java:70)
	at com.google.api.gax.rpc.ClientContext.create(ClientContext.java:168)
	at com.google.cloud.monitoring.v3.stub.GrpcMetricServiceStub.create(GrpcMetricServiceStub.java:191)
	at com.google.cloud.monitoring.v3.stub.MetricServiceStubSettings.createStub(MetricServiceStubSettings.java:406)
	at com.google.cloud.monitoring.v3.MetricServiceClient.<init>(MetricServiceClient.java:169)
	at com.google.cloud.monitoring.v3.MetricServiceClient.create(MetricServiceClient.java:151)
	at io.micrometer.stackdriver.StackdriverMeterRegistry.start(StackdriverMeterRegistry.java:129)
	at io.micrometer.stackdriver.StackdriverMeterRegistry.<init>(StackdriverMeterRegistry.java:114)
	at io.micrometer.stackdriver.StackdriverMeterRegistry.<init>(StackdriverMeterRegistry.java:94)
	at nct.cloud.observe.MetricsStaticObjects.micrometerMetricsOptions(ObservabilityPlugin.kt:103)
	at nct.cloud.observe.MetricsStaticObjects.<clinit>(ObservabilityPlugin.kt:99)
	at nct.cloud.observe.ObservabilityPlugin.configure(ObservabilityPlugin.kt:43)
	at kos.core.Launcher.runPluginsAndConfigureKos(Launcher.java:66)
	at kos.core.Launcher.run(Launcher.java:49)
	at kos.core.Launcher.main(Launcher.java:45)
```

### Steps to reproduce

1. Have a vert.x setup that has Stackdriver Metrics using micrometer, stackdriver tracing using open telemetry and the opencenus shim for open telemetry installed.
2. Start the application

**Comment by vietj** [comment]
@steven-aerts thanks for the info, do you mind creating a pull request to add this valuable information to the documentation ?

**Comment by Traderjoe95** [comment]
> So the so VertxContextStorageProvider will only be used for Vert.x

@steven-aerts, are you sure that this is correct? From a quick look into the source code it seems to me that your suggestion sets the `default` context storage for _everything_. That means that the `VertxContextStorageProvider` isn't used for anything, because there are no direct references to it, it is only used as an SPI, which is loaded by OpenTelemetry if it is requested or if it is the only one on the classpath.

Sure, it fixes the stacktrace this thread is about. But at the same time it will probably break tracing on the Vert.x Event Loop. The default context storage uses `ThreadLocal` to store the context, which is obviously too generic for Vert.x because there are multiple tasks interleaved on the same event loop thread which oftentimes keep spans open across a suspension (e.g. waiting for an HTTP call or something). This means that child spans will be attached incorrectly.

I think a workable solution would rather be what I suggested on #68: Let the `VertxContextStorageProvider` delegate to the default context storage when called on a thread that has no Vert.x Context.

**Comment by tsegismont** [comment]
Fixed by #72
